datameta/validation.py

```diff
@@ -135,11 +139,12 @@ def validate_submission_uniquekeys(db, db_files, db_msets):
         for db_mset in db_msets.values():
             for mdatrec in db_mset.metadatumrecords:
                 if mdatrec.metadatum.name == key:
-                    value_msets[mdatrec.value].append(db_mset)
+                    if mdatrec.value is not None:
+                        value_msets[mdatrec.value].append(db_mset)
         # Reduce to those values that occur in more than one metadatast
-        value_msets = { k: v for k, v in value_msets.items() if len(v) > 1 }
+        not_unique = [ v for v in value_msets.values() if len(v) > 1 ]
```

Comment:
```suggestion
        not_unique = ( v for v in value_msets.values() if len(v) > 1 )
```
No list needed, can use a generator here.